peps/pep-0750.rst

```diff
@@ -0,0 +1,894 @@
+results can be cached/persisted in useful ways.
+
+Tool Support
+============
+
+Python Semantics in Tag Strings
+-------------------------------
+
+Python template languages and other DSLs have semantics quite apart from Python.
+Different scope rules, different calling semantics e.g. for macros, their own
+grammar for loops, and the like.
+
+This means all tools need to write special support for each language. Even then,
+it is usually difficult to find all the possible scopes, for example to autocomplete
+values.
+
+However, f-strings do not have this issue. An f-string is considered part of Python.
+Expressions in curly braces behave as expected and values should resolve based on
+regular scoping rules. Tools such as mypy can see inside f-string expressions,
+but will likely never look inside a Jinja2 template.
+
+DSLs written with tag strings will inherit much of this value. While we can't expect
+standard tooling to understand the "domain" in the DSL, they can still inspect
+anything expressible in an f-string.
+
+Backwards Compatibility
+=======================
+
+Like f-strings, use of tag strings will be a syntactic backwards incompatibility
+with previous versions.
+
+Security Implications
+=====================
+
+The security implications of working with interpolations, with respect to
+interpolations, are as follows:
+
+1. Scope lookup is the same as f-strings (lexical scope). This model has been
+   shown to work well in practice.
+
+2. Tag functions can ensure that any interpolations are done in a safe fashion,
+   including respecting the context in the target DSL.
+
+How To Teach This
+=================
+
+Tag strings have several audiences: consumers of tag functions, authors of tag
+functions, and framework authors who provide interesting machinery for tag
+functions.
+
+All three groups can start from an important framing:
+
+- Existing solutions (such as template engines) can do parts of tag strings
+- But tag strings move everything closer to "normal Python"
+
+Consumers can look at tag strings as starting from f-strings:
+
+- They look familiar
+- Scoping and syntax rules are the same
+- You just need to import the tag function
+
+They first thing they need to absorb: unlike f-strings, the string isn't
+immediately evaluated "in-place". Something else (the tag function) happens.
+That's the second thing to teach: the tag functions do something particular.
+Thus the concept of "domain specific languages" (DSLs).
+
+Tag function authors therefore think in terms of making a DSL. They have
+business policies they want to provide in a Python-familiar way. With tag
+functions, Python is going to do much of the pre-processing. This lowers
+the bar for making a DSL.
+
+Tag authors can start with simple uses. Tag strings can then open to larger
+patterns: lazy evaluation, intermediate representations, registries, and more.
+
+Finally, framework authors can provide contact points with their lifecycles.
```

Comment:
This paragraph is a bit confusing; in particular, "contact points with their lifecycles".

Reply:
Good point, it presumes someone knows about HTML frontend patterns for component driven development. I'll either beef it up or strike it.

Reply:
This will be removed in Jim's next PR update.